File: web/src/main/java/no.nav.sbl/rest/domain/RSContext.java
package no.nav.sbl.rest.domain;

public class RSContext {

    public transient String ip;
    public String aktivBruker;
    public String aktivEnhet;

    public RSContext withAktivBruker(String aktivBruker) {
        this.aktivBruker = aktivBruker;
        return this;
    }

    public RSContext withAktivEnhet(String aktivEnhet) {
        this.aktivEnhet = aktivEnhet;
        return this;
    }

    public RSContext withIP(String ip) {
        this.ip = ip;
        return this;
    }
}

File: src/main/java/no/nav/sbl/rest/model/RSContext.kt
package no.nav.sbl.rest.model

import no.nav.sbl.domain.VeilederContext
import no.nav.sbl.domain.VeilederContextType

data class RSContext(
    var aktivBruker: String? = null,
    var aktivEnhet: String? = null,
) {
    companion object {
        fun from(veilederContext: VeilederContext): RSContext =
            RSContext(
                aktivBruker = if (VeilederContextType.NY_AKTIV_BRUKER == veilederContext.contextType) veilederContext.verdi else null,
                aktivEnhet = if (VeilederContextType.NY_AKTIV_ENHET == veilederContext.contextType) veilederContext.verdi else null,
            )
    }
}
